public/app/features/dashboard-scene/utils/createPanelDataProvider.ts

```diff
@@ -24,6 +24,9 @@ export function createPanelDataProvider(panel: PanelModel): SceneDataProvider |
     dataProvider = new SceneQueryRunner({
       queries: panel.targets,
       maxDataPoints: panel.maxDataPoints ?? undefined,
+      dataLayerFilter: {
+        panelId: panel.id,
+      },
```

Comment:
One potentially tricky bit of logic is that the panelId filtering should only happen when 
* the annotation query has a filter.ids array 
* is of type "dashboard" (should only apply to annotation queries for the build in -- Grafana -- data source)

https://github.com/grafana/grafana/blob/main/public/app/features/query/state/DashboardQueryRunner/utils.ts#L71

One solution for that would be to only return annotation items with panelId when the query has type "dashboard" , and skip them / clear them when type is "tags"

Reply:
@torkelo - yes, this is handled by the transformation imlpemented in https://github.com/grafana/scenes/pull/329

Reply:
nice!